src/app/dev/DevToys.Tools/Tools/Converters/Date/DateConverterGuiTool.cs

```diff
@@ -0,0 +1,1022 @@
+                                    Item(DateConverter.Seconds, DateFormat.Seconds),
+                                    Item(DateConverter.Milliseconds, DateFormat.Milliseconds)
+                                )
+                        )
+                ),
+                Cell(
+                    DateConverterGridRow.Content,
+                    DateConverterGridColumn.Content,
+                    Stack()
+                        .Vertical()
+                        .SmallSpacing()
+                        .WithChildren(
+                            DstInformation(),
+                            _numberInputText
+                                .Title(DateConverter.TimestampTitle)
+                                .OnTextChanged(OnTimeStampChanged)
+                                .Step(1)
+                                .Minimum(-2177452704000000)
+                                // Ticks max value
+                                .Maximum(3155861951990000000),
+                            DateTimeStack()
+                        )
+                )
+            )
+        );
+
+    // Smart detection handler.
+    public void OnDataReceived(string dataTypeName, object? parsedData)
+    {
+        if (dataTypeName == PredefinedCommonDataTypeNames.Date &&
+            parsedData is DateTimeOffset dateStrongTypedParsedData)
+        {
+            _timeYearInputNumber.Value(dateStrongTypedParsedData.Year);
+            _timeMonthInputNumber.Value(dateStrongTypedParsedData.Month);
+            _timeDayInputNumber.Value(dateStrongTypedParsedData.Day);
+            _timeHourInputNumber.Value(dateStrongTypedParsedData.Hour);
+            _timeMinuteInputNumber.Value(dateStrongTypedParsedData.Minute);
+            _timeSecondsInputNumber.Value(dateStrongTypedParsedData.Second);
+            _timeMillisecondsInputNumber.Value(dateStrongTypedParsedData.Millisecond);
+        }
+    }
+
+    public void Dispose()
+    {
+        _cancellationTokenSource?.Dispose();
+        _semaphore.Dispose();
+    }
+
+    private void OnTimeZoneSelected(IUIDropDownListItem? item)
+    {
+        if (item is null)
+        {
+            return;
+        }
+        if (_ignoreInputTextChange)
+        {
+            return;
+        }
+        SelectedTimeZoneDropDownItem = item;
+        _timeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(item.Value!.ToString()!);
+        _settingsProvider.SetSetting(timeZoneIdSettings, _timeZoneInfo.Id);
+        StartDateTimeConvert(_currentTimeUtc);
+    }
+
+    private void OnFormatChanged(DateFormat format)
+    {
+        _dateFormat = format;
+        StartDateTimeConvert(_currentTimeUtc);
+        if (format is not DateFormat.Seconds)
+        {
+            _timeMillisecondsInputNumber.Editable();
+            return;
+        }
+        _timeMillisecondsInputNumber.ReadOnly();
+    }
+
+    private void OnCustomEpochChanged(bool useCustomEpoch)
+    {
+        _useCustomEpoch = useCustomEpoch;
```

Comment:
Could we store the result of `EpochStack` in a field, and `Disable()` / `Enable()` the stack based on whether `useCustomEpoch` is true or false?